common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -0,0 +1,188 @@
+
+    Default to the display_name if it isn't None, else fall back to creating
+    a name based on the URL.
+
+    Unlike the rest of this module's functions, this function takes an entire
+    course descriptor/overview as a parameter. This is because certain test
+    cases (specifically, {Text|Image|Video}AnnotationModuleTestCase.test_student_view)
+    create scenarios where course.display_name is not None but course.location
+    is None, which causes calling course.url_name to fail. So, although we'd
+    like to just pass course.display_name and course.url_name as arguments to
+    this function, we can't do so without breaking those tests.
+
+    Arguments:
+        course (CourseDescriptor|CourseOverview): descriptor or overview of said course.
+    """
+    return (
+        course.display_name if course.display_name is not None
+        else course.url_name.replace('_', ' ')
+    ).replace('<', '&lt;').replace('>', '&gt;')
+
+
+def number_for_course_location(location):
+    """
+    Given a course location, returns the course's number.
+
+    Arguments:
+        location (UsageKey): The location of the course in question.
+    """
+    return location.course
+
+
+def has_course_started(start_date):
+    """
+    Given a course's start datetime, returns whether the current time is past it.
+
+    Arguments:
+        start_date (datetime): The start datetime of the course in question.
+    """
+    return datetime.now(UTC()) > start_date
+
+
+def has_course_ended(end_date):
+    """
+    Given a course's end datetime, returns whether (a) it is not None (b) the current time is past it.
+
+    Arguments:
+        end_date (datetime): The end datetime of the course in question.
+    """
+    return datetime.now(UTC()) > end_date if end_date is not None else False
+
+
+def _add_timezone_string(date_time):
+    """
+    Adds 'UTC' string to the end of start/end date and time texts.
+    """
+    return date_time + u" UTC"
+
+
+def course_start_date_is_default(start, advertised_start):
+    """
+    Returns whether a course's start date hasn't yet been set.
+
+    Arguments:
+        start (datetime): The start datetime of the course in question.
+        advertised_start (str): The advertised start date of the course in question.
+    """
+    return advertised_start is None and start == DEFAULT_START_DATE
+
+
+def course_start_datetime_text(start, advertised_start, format_string, ugettext, strftime):
+    """
+    Calculates text to be shown to user regarding a course's start datetime in UTC..
+    Prefers .advertised_start, then falls back to .start.
+
+    Arguments:
+        start (datetime): the start date of a course
+        advertised_start (str): the advertised start date of a course, as a string
+        format_string (str): the date format type, as passed to strftime
+        ugettext (str -> str): a text localization function
+        strftime (datetime, str -> str): a localized string formatting function
```

Comment:
I can't seem to find anything on our doc guideline on passing functions. Action item would be to bring it up in dev.